src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1090,61 +1087,117 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                     ? receiver.Type
                     : this.ContainingType;
                 method = new ErrorMethodSymbol(methodContainer, returnType, name);
-                args = BuildArgumentsForErrorRecovery(analyzedArguments);
             }
 
+            // until we've considered the extension methods, we don't rewrite the arguments.
+            args = extensionMethodsOfSameViabilityAreAvailable
+                ? analyzedArguments.Arguments.ToImmutable()
+                : BuildArgumentsForErrorRecovery(analyzedArguments, methods);
+
             var argNames = analyzedArguments.GetNames();
             var argRefKinds = analyzedArguments.RefKinds.ToImmutableOrNull();
             return BoundCall.ErrorCall(node, receiver, method, args, argNames, argRefKinds, isDelegate, invokedAsExtensionMethod: invokedAsExtensionMethod, originalMethods: methods, resultKind: resultKind);
         }
 
-        private static ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedArguments analyzedArguments, ImmutableArray<ParameterSymbol> parameters)
+        private bool IsUnboundGeneric(MethodSymbol method)
```

Comment:
Please add a doc comment for this function and consider moving it to src\Compilers\CSharp\Portable\Symbols\MethodSymbolExtensions.cs. 
Also, the usage of "Unbound" term in the name feels confusing. The term has specific meaning for generic types and that meaning is not consistent with the meaning used by this function. I would suggest to use "IsUnconstructedGeneric" name instead because it describes precisely what the method does.